### PR TITLE
task-1: ✅ basics of testing

### PR DESCRIPTION
**Pull request by VladOSIO2**
Created a simple Item class, that represents an item in the stock. Covered it with JUnit unit tests. Implemented unit test run with coverage report for each commit in PR

### CODE REVIEW

File: src/main/java/org/stock/item/Item.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   * Weight in grams
+   */
+  private final int weight;
+
+  /**
+   * method for getting the descriptive info about the item's weight
+   * @return item's weight description as String
+   */
+  public String getWeightDescription() {
+    if (weight < 1000) {
```
Comment (viktoriaobukhova) [inline src/main/java/org/stock/item/Item.java:28]: 1000 looks like magic number - please extract it to the constant

File: src/main/java/org/stock/item/Item.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      return weight + "g";
+    }
+    return weight / 1000 + "kg" + weight % 1000 + "g";
```
Comment (viktoriaobukhova) [inline src/main/java/org/stock/item/Item.java:31]: the same for "g"/"kg"- please extract it to the constants

**Review by viktoriaobukhova — CHANGES_REQUESTED** [review]

File: src/main/java/org/stock/item/Item.java (hunk 1 above)
Comment (VladOSIO2) [reply src/main/java/org/stock/item/Item.java:28]: That makes sense. I will extract it to the constant and clean up the code. Thought about making some sample code with some unit test to implement the test run and coverage report for the PRs first, what apparently was a mistake

File: src/main/java/org/stock/item/Item.java (hunk 2 above)
Comment (VladOSIO2) [reply src/main/java/org/stock/item/Item.java:31]: Ok, same as above, will extract all of these magic variables to the constants

**Comment by VladOSIO2** [comment]
After spending some time on learning the Gradle and workflows, fixed the test run & coverage report. Now after each commit, git workflow will run all the tests and report the test coverage for all affected by PR classes. Workflow check will fail, if any of the tests is failing.

**Comment by VladOSIO2** [comment]
Moved the "magic variables" to the proper constants. Although I have a question: despite the first task requires the creation of single class, will that be a better practice to move this responsibility (calculations of the weight & constants used in it) to another class, something like "WeightFormatter"? And calling the weight formatting method inside the Item's corresponding method, providing the Item's weight to WeightFormatter's method

**Comment by viktoriaobukhova** [comment]
> sure, moving the responsibility to another class that would be responsible for some common actions is definately a good practice 



> Moved the "magic variables" to the proper constants. Although I have a question: despite the first task requires the creation of single class, will that be a better practice to move this responsibility (calculations of the weight & constants used in it) to another class, something like "WeightFormatter"? And calling the weight formatting method inside the Item's corresponding method, providing the Item's weight to WeightFormatter's method

**Review by viktoriaobukhova — APPROVED** [review]

**Comment by VladOSIO2** [comment]
Forgot to squash the commits(

**Pull request merged**